DependencyInjection/CompilerPass.php

```
@@ -0,0 +1,42 @@
+<?php
+
+namespace MovingImage\Bundle\VMProApiBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class CompilerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $clientDefinition = $container->getDefinition('vmpro_api.client');
+
+        $logger = $this->getServiceReference($container, $container->getParameter('vm_pro_api_logger'));
+        $cachePool = $this->getServiceReference($container, $container->getParameter('vm_pro_api_cache_pool'));
```

I believe that this would implicitly pass `null` if you configure a service alias in the API client configuration part for which no actual named service or service alias exists. 

Can we throw an exception instead when the configuration parameter is not `null` but no service can be located for it?